src/oid_array.rs

```diff
@@ -0,0 +1,110 @@
+//! Bindings to libgit2's raw git_strarray type
+
+use std::ops::Range;
+
+use oid::Oid;
+use raw;
+use util::Binding;
+
+/// An oid array structure used by libgit2
+///
+/// Some apis return arrays of oids which originate from libgit2. This
+/// wrapper type behaves a little like `Vec<&Oid>` but does so without copying
+/// the underlying Oids until necessary.
+pub struct OidArray {
+    raw: raw::git_oidarray,
+}
+
+/// A forward iterator over the Oids of an array, casted to `&Oid`.
+pub struct Iter<'a> {
+    range: Range<usize>,
+    arr: &'a OidArray,
+}
+
+/// A forward iterator over the bytes of n array.
+pub struct IterBytes<'a> {
+    range: Range<usize>,
+    arr: &'a OidArray,
+}
+
+impl OidArray {
+    /// Returns None if i is out of bounds.
+    pub fn get(&self, i: usize) -> Option<Oid> {
+        if i < self.raw.count as usize {
+            unsafe {
+                let ptr = self.raw.ids.offset(i as isize);
+                Some(Oid::from_raw(ptr))
+            }
+        } else {
+            None
+        }
+    }
+
+    /// Returns None if `i` is out of bounds.
+    pub fn get_bytes(&self, i: usize) -> Option<&[u8]> {
```

Comment:
Er actually, better idea! Perhaps this could implement `Deref<Target=[Oid]>`?

I was gonna say this may want to implement `Index` but with a deref implementation you should get basically everything!